### ISSUE TITLE
Add ignore file for GitOps Run

### ISSUE THREAD

**Issue by opudrovs** [weaveworks/weave-gitops#3484]
When testing gitops run in the podinfo directory with the `gitops beta run .` command, there is a YAML validation error (at the end of the logs):

```
olga@macbook-pro podinfo % /Volumes/Macintosh_HD_Data/_job_ww/_src/weave-gitops/bin/gitops beta run . 
► Checking for a cluster in the kube config ...
► Preparing the cluster for GitOps Run session ...

You can run `gitops beta run --no-session` to disable session management.

If you are running GitOps Run for the first time, it may take a few minutes to download the required images.
GitOps Run session is also required to install Flux components, if it is not installed yet.
You may see Flux installation logs in the next step.

✗ Would you like to install the GitOps Dashboard: 
► Creating GitOps Run session run-master-1abc44f0 in namespace default ...

You may see Flux installation logs again, as it is being installed inside the session.

► Waiting for GitOps Run session run-master-1abc44f0 to be ready ...
✔ Session run-master-1abc44f0 is ready
► Connecting to GitOps Run session run-master-1abc44f0 ...
► Checking for a cluster in the kube config ...
► Explicitly allow GitOps Run on run-master-1abc44f0 context
► Checking if Flux is already installed ...
► Getting Flux version ...
✗ error getting the flux-system namespace
⚠️ Flux is not found: namespaces "flux-system" not found
Running Flux command: /Users/olga/Library/Caches/.gitops/flux/0.37.0/flux install
✚ generating manifests
✔ manifests build completed
► installing components in flux-system namespace
CustomResourceDefinition/alerts.notification.toolkit.fluxcd.io created
... <- removed some flux installation logs
✔ install finished
► Checking namespace gitops-run ...
✔ Created namespace gitops-run
► Checking service gitops-run/run-dev-bucket ...
✔ Created service gitops-run/run-dev-bucket
► Checking deployment gitops-run/run-dev-bucket ...
✔ Created deployment gitops-run/run-dev-bucket
► Waiting for deployment run-dev-bucket to be ready ...
► Port forwarding to pod gitops-run/run-dev-bucket-798dcb894-g7ldd ...
✔ Port forwarding for run-dev-bucket is ready.
► creating HelmRepository flux-system/fluent
► creating HelmRelease flux-system/fluent-bit
► creating HelmRelease flux-system/fluent-bit
► waiting for HelmRelease flux-system/fluent-bit to be ready
✔ HelmRelease flux-system/fluent-bit is ready
► Checking if GitOps Dashboard is already installed ...
► Checking secret run-dev-bucket-credentials ...
✔ Created secret run-dev-bucket-credentials
✔ Secret run-dev-bucket-credentials already existed
► Checking bucket source run-dev-bucket ...
✔ Created source run-dev-bucket
✔ Source run-dev-bucket already existed
► Checking Kustomization run-dev-ks ...
✔ Created Kustomization run-dev-ks
✔ Setup Bucket Source and Kustomization successfully
◎ Press Ctrl+C to stop GitOps Run ...
► 1 change events detected
► Validating files under ./ ...
/Volumes/Macintosh_HD_Data/_job_ww/_src/podinfo/.github/actions/kubeconform/action.yml - failed validation: error while parsing: missing 'kind' key
✗ Validation failed: please review the errors and try again: validation failed
^C► Received interrupt, quitting...
► Removing Fluent Bit HelmRelease flux-system/fluent-bit ...
► Waiting for HelmRelease flux-system/fluent-bit to be deleted...
✔ HelmRelease flux-system/fluent-bit deleted


► Deleting GitOps Run session run-master-1abc44f0 ...
```

@chanwit suggested that, instead of adding ignored files/directories one by one, we could have smth. like a `.gitopsrunignore` file and just add what we want to ignore to it.

@JamWils, WDYT?

**Comment by makkes** [comment]
IMHO GitOps Run should only validate those files that kustomize-controller/helm-controller would take into account. Granted I don't have all the historic context of the validation functionality in GitOps Run, it may be much simpler to re-use Flux's validation logic here.

**Comment by JamWils** [comment]
@makkes can you expand on what you mean by Flux's validation logic?

**Comment by chanwit** [comment]
related to: https://github.com/weaveworks/weave-gitops/issues/3484

**Comment by JamWils** [comment]
Closing, as we will fix this in #3491.
